python/src/trezorlib/cardano.py

```diff
@@ -24,15 +24,136 @@
 
+INCOMPLETE_OUTPUT_ERROR_MESSAGE = "The output is missing some fields"
+
+
+def create_address_parameters(
+    address_type: messages.CardanoAddressType,
+    path_str: str,
+    staking_key_path_str: str = None,
+    staking_key_hash_str: str = None,
+    block_index: int = None,
+    tx_index: int = None,
+    certificate_index: int = None,
+) -> messages.CardanoAddressParametersType:
+    staking_key_path = None
+    staking_key_hash = None
+    certificate_pointer = None
+
+    if not _is_known_address_type(address_type):
+        raise ValueError("Unknown address type")
+
+    if address_type == messages.CardanoAddressType.BASE_ADDRESS:
+        if staking_key_path_str:
+            staking_key_path = tools.parse_path(staking_key_path_str)
+        elif staking_key_hash_str:
+            staking_key_hash = bytes.fromhex(staking_key_hash_str)
+        else:
+            raise ValueError(
+                "Base address requires a staking key path or a staking key hash"
+            )
+    elif address_type == messages.CardanoAddressType.POINTER_ADDRESS:
+        certificate_pointer = create_certificate_pointer(
+            block_index, tx_index, certificate_index
+        )
+
+    return messages.CardanoAddressParametersType(
+        address_type=address_type,
+        path=tools.parse_path(path_str),
+        staking_key_path=staking_key_path,
+        staking_key_hash=staking_key_hash,
+        certificate_pointer=certificate_pointer,
+    )
+
+
+def _is_known_address_type(address_type: messages.CardanoAddressType) -> bool:
+    return (
+        address_type == messages.CardanoAddressType.BYRON_ADDRESS
+        or address_type == messages.CardanoAddressType.BASE_ADDRESS
+        or address_type == messages.CardanoAddressType.POINTER_ADDRESS
+        or address_type == messages.CardanoAddressType.ENTERPRISE_ADDRESS
+        or address_type == messages.CardanoAddressType.REWARD_ADDRESS
+    )
+
+
+def create_certificate_pointer(
+    block_index: int, tx_index: int, certificate_index: int
+) -> messages.CardanoBlockchainPointerType:
+    if block_index is None or tx_index is None or certificate_index is None:
+        raise ValueError("Invalid pointer parameters")
+
+    return messages.CardanoBlockchainPointerType(
+        block_index=block_index, tx_index=tx_index, certificate_index=certificate_index
+    )
+
+
+def create_input(input) -> messages.CardanoTxInputType:
+    if not all(input.get(k) is not None for k in REQUIRED_FIELDS_INPUT):
+        raise ValueError("The input is missing some fields")
+
+    path = input["path"]
+
+    return messages.CardanoTxInputType(
+        address_n=tools.parse_path(path),
+        prev_hash=bytes.fromhex(input["prev_hash"]),
+        prev_index=input["prev_index"],
+    )
+
+
+def create_output(output) -> messages.CardanoTxOutputType:
+    contains_address = output.get("address")
+    # None check needed, because address type can be 0, thus this would be False
```

Comment:
IMHO this comment is redundant as the line below is self-explanatory. It would be unexpected to skip the None check - i.e. assign a non-bool value to a variable that is apparently meant to be boolean

Reply:
Comment removed. Checks against None added.